### PR TITLE
[WIP][Personnal] Flow analysis pooled objects

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
Re Issue #30589. (Test run on *my machine* is about 90 sec quicker than `master`)

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/Analysis/FlowAnalysis/AbstractFlowPass.NestedTypes.vb [hunk 1]
```diff
@@ -33,9 +33,9 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     count -= 1
                 End If
 
-                If count <= Me._path.Length Then
+                If count <= _path.Length Then
```
Comment (CyrusNajmabadi) [inline src/Compilers/VisualBasic/Portable/Analysis/FlowAnalysis/AbstractFlowPass.NestedTypes.vb]: @AdamSpeight2008 why are style changes being made in the compiler?

File: src/Compilers/VisualBasic/Test/Semantic/FlowAnalysis/FlowTestBase.vb [hunk 2]
```diff
@@ -64,8 +64,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.UnitTests
             Return CompileAndGetModelAndSpan(program, Function(binding, startNodes, endNodes) AnalyzeDataFlow(binding, startNodes, endNodes), ilSource, errors)
         End Function
 
-        Protected Function CompileAndAnalyzeControlAndDataFlow(program As XElement, Optional ilSource As XCData = Nothing, Optional errors As XElement = Nothing) As Tuple(Of ControlFlowAnalysis, DataFlowAnalysis)
-            Return CompileAndGetModelAndSpan(program, Function(binding, startNodes, endNodes) Tuple.Create(AnalyzeControlFlow(binding, startNodes, endNodes), AnalyzeDataFlow(binding, startNodes, endNodes)), ilSource, errors)
+        Protected Function CompileAndAnalyzeControlAndDataFlow(
+                                                                program As XElement,
+                                                       Optional ilSource As XCData = Nothing,
+                                                       Optional errors As XElement = Nothing
+                                                              ) As (ControlFlowAnalysis As ControlFlowAnalysis, DataFlowAnalysis As DataFlowAnalysis)
```
Comment (CyrusNajmabadi) [inline src/Compilers/VisualBasic/Test/Semantic/FlowAnalysis/FlowTestBase.vb]: what is happening here?
1. please do not unnecessarily change formatting
2. please don't use formatting styles that are used *nowhere* else in roslyn.

File: src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineRewriter.vb [hunk 3]
```diff
@@ -271,15 +271,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
...
 
-                Return proxy
+                    Return proxy
+                End If
             End If
 
-            ' Variable needs to be hoisted.
-            Dim fieldType = local.Type.InternalSubstituteTypeParameters(typeMap).Type
+                ' Variable needs to be hoisted.
+                Dim fieldType = local.Type.InternalSubstituteTypeParameters(typeMap).Type
```
Comment (CyrusNajmabadi) [inline src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineRewriter.vb]: why was this indented?

File: src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineRewriter.vb [hunk 4]
```diff
@@ -271,15 +271,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 ' We'll create proxies for these variable later:
                 ' TODO: so, we have to check if it is already in or not. See the early impl.
 
-                Debug.Assert(initializers.ContainsKey(local))
-                proxy = CreateByRefLocalCapture(typeMap, local, initializers)
-                nonReusableLocalProxies.Add(local, proxy)
+                If initializers.ContainsKey(local) Then
```
Comment (CyrusNajmabadi) [inline src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineRewriter.vb]: why did this move from an assert to an 'if check?

File: src/Compilers/VisualBasic/Portable/Analysis/MissingRuntimeMemberDiagnosticHelper.vb [hunk 5]
```diff
@@ -10,13 +10,13 @@ Namespace Microsoft.CodeAnalysis
...
-                                                                                    {"Microsoft.VisualBasic.CompilerServices.ProjectData", "Unstructured exception handling"},
-                                                                                    {"Microsoft.VisualBasic.CompilerServices.ProjectData.CreateProjectError", "Unstructured exception handling"}
-                                                                                 }
+        Private ReadOnly s_metadataNames As New Dictionary(Of String, String) _
+            From {{"Microsoft.VisualBasic.CompilerServices.Operators", "Late binding"},
+                    {"Microsoft.VisualBasic.CompilerServices.NewLateBinding", "Late binding"},
+                    {"Microsoft.VisualBasic.CompilerServices.LikeOperator", "Like operator"},
+                    {"Microsoft.VisualBasic.CompilerServices.ProjectData", "Unstructured exception handling"},
+                    {"Microsoft.VisualBasic.CompilerServices.ProjectData.CreateProjectError", "Unstructured exception handling"}
+                 }
```
Comment (CyrusNajmabadi) [inline src/Compilers/VisualBasic/Portable/Analysis/MissingRuntimeMemberDiagnosticHelper.vb]: why was this formatting changed/

File: src/Compilers/VisualBasic/Portable/Analysis/FlowAnalysis/UnassignedVariablesWalker.vb [hunk 6]
```diff
@@ -20,30 +20,32 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             MyBase.New(info, suppressConstExpressionsSupport:=False, trackStructsWithIntrinsicTypedFields:=True)
         End Sub
 
-        Friend Overloads Shared Function Analyze(info As FlowAnalysisInfo) As HashSet(Of Symbol)
-            Dim walker = New UnassignedVariablesWalker(info)
+        Friend Overloads Shared Function Analyze(info As FlowAnalysisInfo) As PooledObjects.PooledHashSet(Of Symbol)
+            Dim walker As New UnassignedVariablesWalker(info)
```
Comment (CyrusNajmabadi) [inline src/Compilers/VisualBasic/Portable/Analysis/FlowAnalysis/UnassignedVariablesWalker.vb]: please don't change style.

**Comment by CyrusNajmabadi** [comment]
I gave up on the review.  There were too many style changes made.  As has been asked in the past, please don't do that as part of a review.

**Comment by AdamSpeight2008** [comment]
@CyrusNajmabadi  I wasn't expecting a review on a `[personal]`, pull request was mainly for the comparison of test times.

File: src/Compilers/VisualBasic/Portable/Analysis/FlowAnalysis/RegionReachableWalker.vb [hunk 7]
```diff
@@ -15,37 +17,39 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
...
-                    startPointIsReachable = False
-                End If
-            Finally
-                walker.Free()
-            End Try
-        End Sub
-
-        Private _regionStartPointIsReachable As Boolean?
-        Private _regionEndPointIsReachable As Boolean?
+            With walker
```
[the remark this answers is not in the document]
Comment (AdamSpeight2008) [reply src/Compilers/VisualBasic/Portable/Analysis/FlowAnalysis/RegionReachableWalker.vb]: This makes the code within it a lot easier to read.

**Comment by CyrusNajmabadi** [comment]
Sorry, i missed hte 'persona' part.  I looked at the labels, didn't see anything, and i saw that a code review request had been sent.

**Comment by AdamSpeight2008** [comment]
@CyrusNajmabadi Could do with a bot check to disable the review triggering on personal.

File: src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineRewriter.vb (hunk 4 above)
Comment (AdamSpeight2008) [reply src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineRewriter.vb]: @CyrusNajmabadi You get 20 test failures, due to the key already being in the collection.
```
 Microsoft.CodeAnalysis.VisualBasic.UnitTests.AsyncTests.Capture_ExceptionInArrayAccess
System.ArgumentException : An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Microsoft.CodeAnalysis.VisualBasic.StateMachineRewriter`1.CaptureLocalSymbol(TypeSubstitution typeMap, LocalSymbol local, ImmutableDictionary`2 initializers) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Lowering\StateMachineRewriter\StateMachineRewriter.vb:line 317
   at Microsoft.CodeAnalysis.VisualBasic.StateMachineRewriter`1.CreateNonReusableLocalProxies(Result captured, Int32& nextFreeHoistedLocalSlot) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Lowering\StateMachineRewriter\StateMachineRewriter.vb:line 181
   at Microsoft.CodeAnalysis.VisualBasic.StateMachineRewriter`1.Rewrite() in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Lowering\StateMachineRewriter\StateMachineRewriter.vb:line 106
   at Microsoft.CodeAnalysis.VisualBasic.AsyncRewriter.Rewrite(BoundBlock body, MethodSymbol method, Int32 methodOrdinal, VariableSlotAllocator slotAllocatorOpt, TypeCompilationState compilationState, DiagnosticBag diagnostics, AsyncStateMachine& stateMachineType) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Lowering\AsyncRewriter\AsyncRewriter.vb:line 100
   at Microsoft.CodeAnalysis.VisualBasic.Rewriter.RewriteIteratorAndAsync(BoundBlock bodyWithoutLambdas, MethodSymbol method, Int32 methodOrdinal, TypeCompilationState compilationState, DiagnosticBag diagnostics, VariableSlotAllocator slotAllocatorOpt, StateMachineTypeSymbol& stateMachineTypeOpt) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Lowering\Rewriter.vb:line 143
   at Microsoft.CodeAnalysis.VisualBasic.Rewriter.LowerBodyOrInitializer(MethodSymbol method, Int32 methodOrdinal, BoundBlock body, SynthesizedSubmissionFields previousSubmissionFields, TypeCompilationState compilationState, Boolean instrumentForDynamicAnalysis, ImmutableArray`1& dynamicAnalysisSpans, DebugDocumentProvider debugDocumentProvider, DiagnosticBag diagnostics, VariableSlotAllocator& lazyVariableSlotAllocator, ArrayBuilder`1 lambdaDebugInfoBuilder, ArrayBuilder`1 closureDebugInfoBuilder, Int32& delegateRelaxationIdDispenser, StateMachineTypeSymbol& stateMachineTypeOpt, Boolean allowOmissionOfConditionalCalls, Boolean isBodySynthesized) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Lowering\Rewriter.vb:line 106
   at Microsoft.CodeAnalysis.VisualBasic.MethodCompiler.LowerAndEmitMethod(MethodSymbol method, Int32 methodOrdinal, BoundBlock block, Binder binderOpt, TypeCompilationState compilationState, DiagnosticBag diagsForCurrentMethod, ProcessedFieldOrPropertyInitializers processedInitializers, SynthesizedSubmissionFields previousSubmissionFields, MethodSymbol constructorToInject, Int32& delegateRelaxationIdDispenser) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Compilation\MethodCompiler.vb:line 1419
   at Microsoft.CodeAnalysis.VisualBasic.MethodCompiler.CompileMethod(MethodSymbol method, Int32 methodOrdinal, Int32& withEventPropertyIdDispenser, Int32& delegateRelaxationIdDispenser, Predicate`1 filter, TypeCompilationState compilationState, ProcessedFieldOrPropertyInitializers processedInitializers, Binder containingTypeBinder, SynthesizedSubmissionFields previousSubmissionFields, MethodSymbol& referencedConstructor) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Compilation\MethodCompiler.vb:line 1236
   at Microsoft.CodeAnalysis.VisualBasic.MethodCompiler.CompileNamedType(NamedTypeSymbol containingType, Predicate`1 filter) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Compilation\MethodCompiler.vb:line 661
   at Microsoft.CodeAnalysis.VisualBasic.MethodCompiler._Closure$__30-0._Lambda$__0() in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Compilation\MethodCompiler.vb:line 503
   at Roslyn.Utilities.UICultureUtilities.<>c__DisplayClass5_0.<WithCurrentUICulture>b__0() in D:\roslyn-AdamSpeight2008\src\Compilers\Core\Portable\InternalUtilities\UICultureUtilities.cs:line 127
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.CodeAnalysis.VisualBasic.MethodCompiler.WaitForWorkers() in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Compilation\MethodCompiler.vb:line 313
   at Microsoft.CodeAnalysis.VisualBasic.MethodCompiler.CompileMethodBodies(VisualBasicCompilation compilation, PEModuleBuilder moduleBeingBuiltOpt, Boolean emittingPdb, Boolean emitTestCoverageData, Boolean hasDeclarationErrors, Predicate`1 filter, DiagnosticBag diagnostics, CancellationToken cancellationToken) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Compilation\MethodCompiler.vb:line 226
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicCompilation.CompileMethods(CommonPEModuleBuilder moduleBuilder, Boolean emittingPdb, Boolean emitMetadataOnly, Boolean emitTestCoverageData, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Compilation\VisualBasicCompilation.vb:line 2318
   at Microsoft.CodeAnalysis.Compilation.Emit(Stream peStream, Stream metadataPEStream, Stream pdbStream, Stream xmlDocumentationStream, Stream win32Resources, IEnumerable`1 manifestResources, EmitOptions options, IMethodSymbol debugEntryPoint, Stream sourceLinkStream, IEnumerable`1 embeddedTexts, CompilationTestData testData, CancellationToken cancellationToken) in D:\roslyn-AdamSpeight2008\src\Compilers\Core\Portable\Compilation\Compilation.cs:line 2244
   at Roslyn.Test.Utilities.RuntimeEnvironmentUtilities.EmitCompilationCore(Compilation compilation, IEnumerable`1 manifestResources, DiagnosticBag diagnostics, CompilationTestData testData, EmitOptions emitOptions) in D:\roslyn-AdamSpeight2008\src\Test\Utilities\Portable\Compilation\IRuntimeEnvironment.cs:line 216
   at Roslyn.Test.Utilities.RuntimeEnvironmentUtilities.EmitCompilation(Compilation compilation, IEnumerable`1 manifestResources, List`1 dependencies, DiagnosticBag diagnostics, CompilationTestData testData, EmitOptions emitOptions) in D:\roslyn-AdamSpeight2008\src\Test\Utilities\Portable\Compilation\IRuntimeEnvironment.cs:line 194
   at Roslyn.Test.Utilities.Desktop.DesktopRuntimeEnvironment.Emit(Compilation mainCompilation, IEnumerable`1 manifestResources, EmitOptions emitOptions, Boolean usePdbForDebugging) in D:\roslyn-AdamSpeight2008\src\Test\Utilities\Portable\Platform\Desktop\DesktopRuntimeEnvironment.cs:line 200
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationVerifier.Emit(IRuntimeEnvironment testEnvironment, IEnumerable`1 manifestResources, EmitOptions emitOptions) in D:\roslyn-AdamSpeight2008\src\Test\Utilities\Portable\CompilationVerifier.cs:line 138
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationVerifier.Emit(String expectedOutput, Nullable`1 expectedReturnCode, String[] args, IEnumerable`1 manifestResources, EmitOptions emitOptions, Verification peVerify, SignatureDescription[] expectedSignatures) in D:\roslyn-AdamSpeight2008\src\Test\Utilities\Portable\CompilationVerifier.cs:line 103
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.Emit(Compilation compilation, IEnumerable`1 dependencies, IEnumerable`1 manifestResources, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, Action`1 assemblyValidator, Action`1 symbolValidator, EmitOptions emitOptions, Verification verify) in D:\roslyn-AdamSpeight2008\src\Test\Utilities\Portable\CommonTestBase.cs:line 154
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.CompileAndVerifyCommon(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 assemblyValidator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, EmitOptions emitOptions, Verification verify) in D:\roslyn-AdamSpeight2008\src\Test\Utilities\Portable\CommonTestBase.cs:line 68
   at Microsoft.CodeAnalysis.VisualBasic.UnitTests.BasicTestBase.CompileAndVerify(XElement source, IEnumerable`1 allReferences, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, VisualBasicCompilationOptions options, VisualBasicParseOptions parseOptions, EmitOptions emitOptions, Verification verify) in D:\roslyn-AdamSpeight2008\src\Compilers\Test\Utilities\VisualBasic\BasicTestBase.vb:line 179
   at Microsoft.CodeAnalysis.VisualBasic.UnitTests.BasicTestBase.CompileAndVerify(XElement source, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, MetadataReference[] references, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, VisualBasicCompilationOptions options, VisualBasicParseOptions parseOptions, EmitOptions emitOptions, Verification verify, Boolean useLatestFramework) in D:\roslyn-AdamSpeight2008\src\Compilers\Test\Utilities\VisualBasic\BasicTestBase.vb:line 137
   at Microsoft.CodeAnalysis.VisualBasic.UnitTests.AsyncTests.Capture_ExceptionInArrayAccess() in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Test\Emit\CodeGen\CodeGenAsyncTests.vb:line 3608
```

File: src/Compilers/VisualBasic/Test/Semantic/FlowAnalysis/FlowTestBase.vb (hunk 2 above)
Comment (AdamSpeight2008) [reply src/Compilers/VisualBasic/Test/Semantic/FlowAnalysis/FlowTestBase.vb]: Using a tuple literals. Change styling back.

**Comment by jaredpar** [comment]
The PR system is for testing changes that are potential candidates for being merged into the source tree. Personal review is an appropriate designation when working on more complex changes. At the same time PRs with large code style changes don't belong here, even under the personal review tag. Such changes have zero chance of being merged into our code base. 

PRs with large code style changes, no matter how they are labeled, will be summarily closed without comment.

I'm beyond tired of having this discussion with you @AdamSpeight2008. If you continue to send PRs of this nature we will consider more extreme actions such as banning your account from this repository. I really don't want to do this. But you've ignored the many, many requests we've made in this area. I'm not sure what other options we have left here.

**Pull request closed without merging**